### DISCUSSION TITLE
Internet Explorer compatibility?

### DISCUSSION THREAD

**Discussion by msteele999** [ruffle-rs/ruffle#3397] [Q&A]
I have an application (HP Virtual Connect Manager) that requires ActiveX and Flash. So far, only Internet Explorer gets past the ActiveX requirement, but I don't know how (or if I can) load ruffle as an extension in the older Internet Explorer.

Thank you.

**Comment by tsunamistate** [comment]
According to [MDN docs](https://developer.mozilla.org/en-US/docs/WebAssembly#browser_compatibility), Internet Explorer doesn't support WebAssembly (the key technology that Ruffle uses in web browser for emulating Flash).
So, it's unlikely that Ruffle will support IE

**Reply by msteele999** [reply]
understood and thanks! I'm trying to go the other way now - adding activeX support to Chrome / Firefox / Edge,

Problem is that when you run in IE compatibility mode with activeX, you lose the Ruffle plugin ....

**Reply by tsunamistate** [reply]
ActiveX is Internet Explorer exclusive (other browsers basically open IE window within themselves to support it) and has been unsupported from 2015
Unfortunately, it looks like you're out of luck on this one :(

**Reply by danielhjacobs** [reply]
If your application runs on the web, you can still run Flash itself in IE. However, to use it on Windows, you need to get around the Windows kill-switch. This will only work for you or anyone else who does it though, not everyone else like Ruffle would. Create a file called mms.cfg, and move it to C:\Windows\SysWOW64\Macromed\Flash and C:\Windows\System32\Macromed\Flash. I know whenever someone says to mess with System32, you assume it may break your computer (for good reason), but this actually needs to be done. I got a website running Flash that didn't work with Ruffle to work on IE for me this way. Note that IE is already basically unsupported, but due to an agreement Windows made with users when they shipped IE11 with their systems, the browser itself will exist and receive critical security updates for as long as the version of Windows 10 on which it is shipped is supported.

The contents of mms.cfg should be:

EOLUninstallDisable=1
DisableAnalytics=1
SilentAutoUpdateEnable=0
EnableAllowList=1
AllowListPreview=1
AutoUpdateDisable=1
ErrorReportingEnable=1
EnableInsecureActiveXNavigateToURL=1
AllowListRootMovieOnly=1
EnableInsecureLocalWithFileSystem=1
EnableInsecureAllowListLocalPathMatching=1

AllowListUrlPattern=<full_web_url_of_website>

**Comment by msteele999** [comment]
Success!

**Comment by msteele999** [comment]
question - what is the syntax of AllowListUrlPattern for listing multiple
URL's?

space  or comma between them?

**Comment by msteele999** [comment]
I figured it out.

A new line for each one workd
